### ISSUE TITLE
Run analysis on CIViC + MOAlmanac data

### ISSUE THREAD

**Issue by korikuzma** [cancervariants/variation-normalization#287]
Will use [REST API](https://normalize.cancervariants.org/variation/normalize) for this analysis. See what we can and can't normalize using SPDI, HGVS, + free text queries from these sources.

**Comment by wesleygoar** [comment]
@korikuzma Here is the link for the categories document that I worked on for identifying the gap in our normalizer service.  https://docs.google.com/document/d/1OrhHBxg6KuzkagrKMmr0PINFxkxHaogF6qVqMIkO3T0/edit?usp=sharing 
Please let me know if you have questions about this doc. Most of these are key phrases that you had already included in your keywords from before (though some are expanded). I would like these binned into the 9 major categories to start with. If we could get this run on CIViC & MOAlmanac that would be great.

**Comment by korikuzma** [comment]
@wesleygoar When do you want this by?

**Comment by wesleygoar** [comment]
Sometime early next week if possible.

**Comment by korikuzma** [comment]
@wesleygoar where did you get `Double Ph` and `Alu insertion`?

**Comment by wesleygoar** [comment]
https://civicdb.org/molecular-profiles/2878/summary
https://civicdb.org/variants/709/summary

**Comment by korikuzma** [comment]
@wesleygoar did you want me to do molecular profiles? I was just going to do variants

**Comment by korikuzma** [comment]
@wesleygoar some other questions. 

1. What's the difference between an `internal tandem duplication` and a `tandem repeat`? 
2. Is it okay if I just do `homozygosity`? I don't see `"mutation" homozygosity` in variants
3. What category for `{GENE} deletion`?
4. What category for `exon {EXON_NUMBER} mutation`?
5. There is one (which means there might be others) [variant](https://civicdb.org/variants/3204/summary) that is in two categories - fusion and copy number. Should I place in both? 
6. Where should I put `Non-P-Loop Mutation`? 
7. Where should I put `P-Loop Mutation`?

**Comment by korikuzma** [comment]
> By molecular profiles do you mean categorical variation?

So CIViC just added [molecular profiles](https://docs.civicdb.org/en/latest/model/molecular_profiles.html) in v3. In civicpy, I was just going to use `get_all_variants()`. They also have `get_all_molecular_profiles()`. Let me know if you want this analysis too or not.

**Comment by wesleygoar** [comment]
1. Bin those in the same category - rearrangements
2. Yes. "mutation" was for any mutation. Ex. https://civicdb.org/variants/740/summary 
3. Functional - gene function
4. Region defined
5. Fusion - just take whatever is listed first
6. Region defined
7. Region defined

**Comment by korikuzma** [comment]
> > By molecular profiles do you mean categorical variation?
> 
> So CIViC just added [molecular profiles](https://docs.civicdb.org/en/latest/model/molecular_profiles.html) in civicpy v3. In civicpy, I was just going to use `get_all_variants()`. They also have `get_all_molecular_profiles()`. Let me know if you want this analysis too or not.

@ahwagner said just doing an analysis on variants is fine

**Comment by korikuzma** [comment]
@wesleygoar for cases where the CIViC names contains both protein and cDNA change [C162R (c.484T>C) ](https://civicdb.org/variants/1772/summary), did you want me to run both protein and cDNA? Variation Normalizer does not support queries that have both

**Comment by wesleygoar** [comment]
@korikuzma I would just do the cDNA only probably. I dont want to double dip variants

**Comment by korikuzma** [comment]
@wesleygoar are you sure? I can do two separate analysis

**Comment by wesleygoar** [comment]
If we do two separate analyses that is fine. I just don't want to double dip by counting a variant twice in two categories during the same analysis.

**Comment by ahwagner** [comment]
In CIViC, when you see a cDNA change in the label, use the representative variant (genomic) data to normalize.

**Comment by korikuzma** [comment]
@wesleygoar Here is an example of variant ("feature") in MOA. We can see that sometimes there are protein, cDNA, and genomic changes. In [this](https://moalmanac.org/assertion/300) example, it appears the protein representation is what's being used in the assertion. In these cases, should I try to normalize via protein representation? 

`curl -X 'GET' 'https://moalmanac.org/api/features/300' -H 'accept: application/json'`

```json
{
  "attributes": [
    {
      "alternate_allele": "T",
      "cdna_change": "c.1938C>A",
      "chromosome": "7",
      "end_position": "148508726",
      "exon": "16",
      "feature_type": "somatic_variant",
      "gene": "EZH2",
      "protein_change": "p.Y646*",
      "reference_allele": "G",
      "rsid": null,
      "start_position": "148508726",
      "variant_annotation": "Nonsense"
    }
  ],
  "feature_id": 300,
  "feature_type": "somatic_variant"
}
```

**Comment by korikuzma** [comment]
@wesleygoar Initial analysis for CIViC can be found [here](https://github.com/cancervariants/variation-normalization/blob/issue-287/notebooks/analysis/civic/civic_analysis.ipynb). Still waiting on you to look over [this](https://gist.github.com/korikuzma/82f6cea2b9fa070dd07c747f56db940f#file-civic_protein_not_supported-txt) gist so we can update the categories doc. Let me know if you want any additional things to be included in the analysis.

**Comment by korikuzma** [comment]
More comments on initial civic analysis...

- There were some duplicate variants. I've flagged these in CIViC.  
  ```json
    {"NC_000003.11:g.10188243T>C": [2034, 2508],
     "NC_000003.11:g.10183725C>G": [1751, 1787]}
  ```
- There was a typo in an amplification civic query (`MITF Amplificatioin`). I've made a revision in CIViC.
- We're unable to normalize 325 variants, but the `unable_to_normalize_queries` only has a length of 266. I'm looking into what is happening. This does not happen in any of our other files.

**Comment by korikuzma** [comment]
> More comments on initial civic analysis...
> 
>     * There were some duplicate variants. I've flagged these in CIViC.
>       ```json
>         {"NC_000003.11:g.10188243T>C": [2034, 2508],
>          "NC_000003.11:g.10183725C>G": [1751, 1787]}
>       ```
> 
>     * There was a typo in an amplification civic query (`MITF Amplificatioin`). I've made a revision in CIViC.
> 
>     * We're unable to normalize 325 variants, but the `unable_to_normalize_queries` only has a length of 266. I'm looking into what is happening. This does not happen in any of our other files.

Fixed the `unable_to_normalize_queries`

**Comment by korikuzma** [comment]
> In CIViC, when you see a cDNA change in the label, use the representative variant (genomic) data to normalize.

We will use either the representative variant data (to make gnomad vcf format) or look for genomic HGVS expression.

In cases where neither of this data exists, our next step would be to normalize on the pattern `<Gene> <HGVS c-like>`. We would have to find the best compatible transcript for the provided gene variant, and return the associated genomic variant reverse translating from c. to g. This is not a priority.

**Comment by korikuzma** [comment]
@wesleygoar we should be able to support `EGFR L747_T751delLREAT`fairly easily. It is similar to #328 , but for deletions

**Comment by wesleygoar** [comment]
@ahwagner tagging you for the above comment

**Comment by korikuzma** [comment]
> @ahwagner tagging you for the above comment

@ahwagner just want to see if it's okay if I work on this easy issue

**Comment by ahwagner** [comment]
Go for it @korikuzma

**Comment by korikuzma** [comment]
From @ahwagner on variant --> MP --> EIDs status analysis
> The intent of this analysis is to address the question of:
> How well do variants (MPs) normalize, comparing those that have been reviewed and approved by a CIViC editor against those that have not?
> So I would say if a variant has at least one EID in an accepted status, it counts towards “accepted”, because that indicates review and approval of the variant as part of the evidence review.

**Comment by korikuzma** [comment]
TODO: Re-run once #394 is merged. Shouldn't impact anything since we don't store VRS IDs anywhere, but still good to re-run after a while.

**Comment by korikuzma** [comment]
@wesleygoar @anastasiasmith1221 re-ran civic analysis in [this](https://github.com/cancervariants/variation-normalization/commit/c88917228480b7791e91d18b11ac6d3f327c3598) commit.

**Comment by korikuzma** [comment]
@wesleygoar Went ahead and re-ran MOA just for kicks in [this](https://github.com/cancervariants/variation-normalization/commit/0420eed0718869f391f5d7720b3f826a2a06d12d) commit. Pull from `issue-287` to see the latest changes.

**Comment by korikuzma** [comment]
Raised a flag for the civic duplicate variant [here](https://civicdb.org/variants/3452/flags)

**Comment by korikuzma** [comment]
TODO: Re-run CIViC analysis. `Single nucleotide variant` will be classified as `region-defined variant`

**Comment by korikuzma** [comment]
> TODO: Re-run CIViC analysis. `Single nucleotide variant` will be classified as `region-defined variant`

Done in latest commit

**Comment by korikuzma** [comment]
@wesleygoar can you let me know if anything else needs to be done for CIViC or MOA?

**Comment by korikuzma** [comment]
TODO: It might be best to combine transcript + not supported csv files since transcript is a subset of not supported

**Comment by wesleygoar** [comment]
@korikuzma agreed.

**Comment by korikuzma** [comment]
> TODO: It might be best to combine transcript + not supported csv files since transcript is a subset of not supported

@anastasiasmith1221 @wesleygoar pushed in latest commit

**Comment by korikuzma** [comment]
Did not mean to close
